### PR TITLE
Openid connect

### PR DESCRIPTION
**Pull request by shane-tomlinson**

### CODE REVIEW

File: bin/persona-gmail-bridge.js [hunk 1]
```diff
@@ -280,43 +263,65 @@ app.get('/authenticate/forward', validate({ email: 'gmail' }),
...
-  // Check input precondition:
-  // OpenID params must include only one email address, and it must be signed.
-  if (!oidTool.validParams(req.query)) {
-    return res.status(401).render('error',
-      { title: req.gettext('Error'), errorInfo: 'Email not signed.' });
-  }
-
-  openidRP.verifyAssertion(req, function (error, result) {
+  google.tradeCodeForEmail(code, function (error, result) {
     if (error && error.message === 'Authentication cancelled') {
```
Comment (shane-tomlinson) [inline bin/persona-gmail-bridge.js]: We should safely be able to remove this branch. We were previously passing in the request to `verifyAssertion`, which would check for errors and surface them.

File: lib/google.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+          return callback(error);
+        }
+
+        if (! verifiedEmail) {
+          return callback(new Error('Could not get email'));
+        }
+
+        callback(null, {
+          email: verifiedEmail,
+          authenticated: true
```
Comment (shane-tomlinson) [inline lib/google.js]: The authenticated field here and the check [here](https://github.com/mozilla/persona-gmail-bridge/pull/131/files#diff-6c82efa6824e9c1d6a9f6ffa88cb2589R309) do not really add any value - we are passing back a `verifiedEmail` already, which indicates the user is indeed authenticated.

Perhaps we can change the `email` field here to `verifiedEmail` which better implies its purpose?

Comment (shane-tomlinson) [inline lib/google.js]: Actually, if we are trading a code for an email, it means we have a code, which implies the user is authenticated & verified by google. Maybe just remove the `authenticated` portion.

**Comment by shane-tomlinson** [comment]
> @shane-tomlinson You broke the tests, dude! ;-) TypeError: Object #<Object> has no method 'setCredentials'

Doh. FWIW, I came up with the same solution to fix the tests, character for character.

**Comment by gene1wood** [comment]
In testing this in stage it looks good. Only thing I noticed is UX which is that the window size doesn't fit well with the current Google login page UI. It maybe have been like this before and it's probably not a big deal. Persona popup

**Comment by shane-tomlinson** [comment]
:smile:

**Comment by groovecoder** [comment]
:heart_eyes:

**Pull request merged**